bundle/config/variable/variable.go

```diff
@@ -2,12 +2,27 @@ package variable
 
 import (
 	"fmt"
+	"reflect"
+)
+
+// We are using `any` because since introduction of complex variables,
+// variables can be of any type.
+// Type alias is used to make it easier to understand the code.
+type VariableValue = any
+
+type VariableType string
+
+const (
+	VariableTypeComplex VariableType = "complex"
 )
 
 // An input variable for the bundle config
 type Variable struct {
+	// A type of the variable. This is used to validate the value of the variable
+	Type VariableType `json:"type,omitempty"`
+
```

Comment:
Would we want to constrain the type of primitive types as well?

And otherwise, could we get away with not storing this at all?

Reply:
We indeed could get away with not having this but there are few benefits of having this
1. It makes the intention of using complex variable explicit
2. It allows for visual editing tools to easily identify that the variable is of complex type and disable editing
3. It give the ability for future to extend the type with a known types, f.e. `job_clusters` and etc. I can imagine restricting the primitive types can be beneficial as well